tasks/componentUpdateTasks.ts

```diff
@@ -0,0 +1,186 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as gulp from 'gulp';
+import * as process from 'node:process';
+import * as fs from 'fs';
+import * as path from 'path';
+import minimist from 'minimist';
+import { spawnSync } from 'child_process';
+import { getPackageJSON } from './packageJson';
+import { Octokit } from '@octokit/rest';
+
+type Options = {
+    userName?: string;
+    email?: string;
+};
+
+async function git(args: string[], printCommand = true): Promise<string> {
+    if (printCommand) {
+        console.log(`git ${args.join(' ')}`);
+    }
+
+    const git = spawnSync('git', args);
+    if (git.status != 0) {
+        const err = git.stderr ? git.stderr.toString() : '';
+        if (printCommand) {
+            console.error(`Failed to execute git ${args.join(' ')}.`);
+        }
+        throw new Error(err || `git ${args.join(' ')} failed with code ${git.status}`);
+    }
+
+    const stdout = git.stdout ? git.stdout.toString() : '';
+    if (printCommand) {
+        console.log(stdout);
+    }
+    return Promise.resolve(stdout);
+}
+
+gulp.task('publish roslyn copilot', async () => {
+    const parsedArgs = minimist<Options>(process.argv.slice(2));
+
+    // Get staging directory from environment variable passed from pipeline
+    const stagingDir = process.env['STAGING_DIRECTORY'];
+    if (!stagingDir) {
+        console.log('STAGING_DIRECTORY environment variable not set; skipping package.json update.');
+        return;
+    }
+
+    if (!fs.existsSync(stagingDir)) {
+        console.log(`Staging directory not found at ${stagingDir}; skipping package.json update.`);
+        return;
+    }
+
+    // Find the Roslyn zip file in the staging directory (we know it was copied here)
+    const files = fs.readdirSync(stagingDir);
+    const zipFile = files.find((file) => /Roslyn\.LanguageServer.*\.zip$/i.test(file));
+
+    if (!zipFile) {
+        console.log(`No Roslyn LanguageServer zip file found in ${stagingDir}; skipping package.json update.`);
+        return;
+    }
+
+    const zipPath = path.join(stagingDir, zipFile);
+    console.log(`Using zip file: ${zipPath}`);
+    const zipName = zipFile;
+
+    // Extract version from file name
+    // Example: "Microsoft.VisualStudio.Copilot.Roslyn.LanguageServer-18.0.671-alpha.zip"
+    let version: string | null = null;
+    const m = zipName.match(/Microsoft\.VisualStudio\.Copilot\.Roslyn\.LanguageServer-(.+)\.zip$/i);
+    if (m && m[1]) {
+        version = m[1];
+    }
+
+    if (!version) {
+        console.log(`Could not extract version from file name ${zipName}; skipping.`);
+        return;
+    }
+
+    console.log(`Extracted version: ${version}`);
```

Comment:
I would check here whether the PR exists and exit if so.

Then I would check if the branch exists on the remote and create it with the insertion if not.

Then finish by creating the PR for the existing or newly pushed branch.